pkg/reconciler/v1alpha1/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -200,6 +206,15 @@ func (e *ResourceNotFoundError) Error() string {
 	return fmt.Sprintf("Couldn't retrieve PipelineResource: %s", e.Msg)
 }
 
+type ConditionNotFoundError struct {
+	Name string
+	Msg  string
+}
+
+func (e *ConditionNotFoundError) Error() string {
+	return fmt.Sprintf("Couldn't retrieve Condition %q: %s", e.Name, e.Msg)
+}
```

Comment:
another thought tho: I noticed you moved the condition resolution logic into its own file with it's own tests - instead of having `ResolvePipelineRun` call `resolveConditionChecks`, the reconciler could call `resolveConditionChecks` directly - maybe the problem we're having with returning a specific error could indicate this function has too many responsibilities?

(It would also reduce the number of test cases you need to test at the `ResolvePipelineRun` level!)

Reply:
The main reason for adding the `ConditionNotFound` error type (besides tests) is to return an error when a PipelineRun fails due to a missing Condition: https://github.com/tektoncd/pipeline/pull/1093/files#diff-225ed18b17b573cba99d592dea1da1a0R290

I was following the pattern there where we return different error types for Resources or Tasks not being found.

Reply:
`ResolvePipelineRun` definitely does a lot (e.g. resolving tasks, taskruns resources, and now conditions and checks). I'm open to ideas on simplifying/modularizing this. At the moment though, we need this in order to build the full `PipelineRunState` in one call which is then used by the reconciler.

Reply:
> ResolvePipelineRun definitely does a lot (e.g. resolving tasks, taskruns resources, and now conditions and checks). I'm open to ideas on simplifying/modularizing this. At the moment though, we need this in order to build the full PipelineRunState in one call which is then used by the reconciler.

kk maybe we can refactor it in follow up PRs! For me the most appealing option is to have multiple `Resolve` functions, each resolving a different type of thing, each being called by the Reconciler - then the Reconciler will know what kind of error message to set cuz it will know what kind of thing its trying to reconcile. Not super important tho!